src/wireviz/DataClasses.py

```diff
@@ -55,6 +59,18 @@ def __post_init__(self):
         if len(self.pinnumbers) != len(set(self.pinnumbers)):
             raise Exception('Pin numbers are not unique')
 
+        if self.show_name is None:
+            if self.autogenerate:
+                self.show_name = False  # hide auto-generated designators by default
+            else:
+                self.show_name = True
+
+        if self.show_pincount is None:
+            if self.style == 'simple':
+                self.show_pincount = False  # hide pincount for simple connectors, since they are 1 pin connectors by definition
+            else:
+                self.show_pincount = True
```

Comment:
The if-else spanning lines 69-72 might be replaced by:
`self.show_pincount = self.style != 'simple'  # hide pincount for simple (1 pin) connectors by default`

Reply:
Done.